models/intermediate/account_balances/int_account_balances__token_transfers.sql

```diff
@@ -0,0 +1,147 @@
+        , "granularity": "day"
+    },
+    "cluster_by": ["account_id", "contract_id"],
+    "incremental_predicates": ["DBT_INTERNAL_DEST.day >= DATE_SUB(DATE('" ~ var('execution_date') ~ "'), INTERVAL 1 DAY)"]
+} %}
+
+{{ config(
+    meta=meta_config,
+    **meta_config,
+    )
+}}
+
+-- This table computes the daily balance for a given account_id
+-- Note that account_id currently only includes C addresses because int_account_balances__token_transfers_value_movement
+-- currently only includes C addresses
+with
+{% if is_incremental() %}
+        -- Create a union of previous balances and new value movements for incremental runs
+        -- so we don't have to aggregate the entire history again
+        previous_balances as (
+            select
+                day
+                , account_id
+                , contract_id
+                , balance
+            from {{ this }}
+            where true
+                -- Get the latest balance from before the incremental run date
+                and day < date('{{ var("batch_end_date") }}')
+            qualify row_number() over (
+                partition by account_id, contract_id
+                order by day desc
+            ) = 1
+        ),
+
+        value_movements as (
+            select
+                day
+                , account_id
+                , contract_id
+                , balance
+            from {{ ref('int_account_balances__token_transfers_value_movement') }}
+            where true
+                -- Get the value movements since the last balance date
+                and day > (select max(day) from previous_balances)
+                and day < date('{{ var("batch_end_date") }}')
+        ),
+
+        merged_value_movements as (
+            select * from previous_balances
+            union all
+            select * from value_movements
+        ),
+    {% endif %}
+
+    account_date_ranges as (
+        select
+            account_id
+            , contract_id
+            , min(day) as start_day
+            , max(day) as last_change_day
+        {% if is_incremental() %}
+                from merged_value_movements
+            {% else %}
+            from {{ ref('int_account_balances__token_transfers_value_movement') }}
+        {% endif %}
+        where
+            true
+            {% if is_incremental() %}
+            and day < date('{{ var("batch_end_date") }}')
+        {% else %}
+                and day <= date('{{ var("batch_start_date") }}')
+            {% endif %}
+        group by 1, 2
+    )
+
+    , date_spine as (
+        select
+            adr.account_id
+            , adr.contract_id
+            , day
+        from account_date_ranges as adr
+        {% if is_incremental() %}
+            , unnest(generate_date_array(adr.start_day, date('{{ var("batch_end_date") }}'))) as day
+        {% else %}
+            , unnest(generate_date_array(adr.start_day, date('{{ var("batch_start_date") }}'))) as day
+        {% endif %}
```

Comment:
maybe a hot take: I think this the date date array should just be batch_start_date, batch_end_date
I imagine we're going to backfill this exactly like we backfill entity tables. I don't think we'd ever fully refresh the tables in one go. Has benchmarking proven otherwise?